runtime/src/utils/buffer/write.rs

```diff
@@ -35,15 +114,9 @@ use crate::{Blob, Error};
 ///     assert_eq!(&buf, b"hello world!");
 /// });
 /// ```
+#[derive(Clone)]
 pub struct Write<B: Blob> {
-    /// The underlying blob to write to.
-    blob: B,
-    /// The buffer storing data to be written to the blob.
-    buffer: Vec<u8>,
-    /// The current position in the blob where the next flush will write.
-    position: u64,
-    /// The maximum size of the buffer.
-    capacity: usize,
+    inner: Arc<RwLock<Inner<B>>>,
```

Comment:
While it is a shame we now have a lock here, the benefits of having a write buffer far exceed the downsides.